indico/modules/events/registration/forms.py

```diff
@@ -524,5 +508,22 @@ class RegistrationPrivacyForm(IndicoForm):
                                                                 'to everyone else who can access the event'))
     submitted = HiddenField()
 
+    def __init__(self, *args, **kwargs):
+        self.regform = kwargs.pop('regform')
+        super().__init__(*args, **kwargs)
+
     def is_submitted(self):
         return super().is_submitted() and 'submitted' in request.form
+
+    def validate_visibility(self, field):
+        participant_visibility, public_visibility = (PublishRegistrationsMode[v] for v in field.data)
+        if participant_visibility.value < public_visibility.value:
+            raise ValidationError(_('Registrations list visibility can not be more restrictive to participants than '
+                                    'to the public'))
+        if (self.regform and
+                self.regform.existing_registrations_count > 0 and
+                ((participant_visibility == PublishRegistrationsMode.show_all and
+                  self.regform.publish_registrations_participants != PublishRegistrationsMode.show_all) or
+                 (public_visibility == PublishRegistrationsMode.show_all and
+                  self.regform.publish_registrations_public != PublishRegistrationsMode.show_all))):
```

Comment:
I don't think this block is particularly readable...

Reply:
I agree... Would this be more readable?
```python
participant_visibility_changed_to_show_all = (participant_visibility == PublishRegistrationsMode.show_all and
                                                self.regform.publish_registrations_participants !=
                                                PublishRegistrationsMode.show_all)
public_visibility_changed_to_show_all = (public_visibility == PublishRegistrationsMode.show_all and
                                            self.regform.publish_registrations_public !=
                                            PublishRegistrationsMode.show_all)
if (self.regform and
        self.regform.existing_registrations_count > 0 and
        (participant_visibility_changed_to_show_all or public_visibility_changed_to_show_all)):
```
Or this?

```python
old_participant_visibility = self.regform.publish_registrations_participants
old_public_visibility = self.regform.publish_registrations_public
if (self.regform and
        self.regform.existing_registrations_count > 0 and
        ((participant_visibility == PublishRegistrationsMode.show_all and
            old_participant_visibility != PublishRegistrationsMode.show_all) or
            (public_visibility == PublishRegistrationsMode.show_all and
            old_public_visibility != PublishRegistrationsMode.show_all))):
```

Reply:
the second one is just as bad as the original IMHO, but how about this (based on your first one):

```python
participant_visibility_changed_to_show_all = (
    participant_visibility == PublishRegistrationsMode.show_all and
    self.regform.publish_registrations_participants != PublishRegistrationsMode.show_all
)
public_visibility_changed_to_show_all = (
    public_visibility == PublishRegistrationsMode.show_all and
    self.regform.publish_registrations_public != PublishRegistrationsMode.show_all
)

if (
    self.regform and
    self.regform.existing_registrations_count > 0 and
    (participant_visibility_changed_to_show_all or public_visibility_changed_to_show_all)
):
    ...
```

That way you avoid the ugly wrapping within a binary expression